packages/loader/container-loader/src/container.ts

```diff
@@ -1648,20 +1646,27 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     /**
      * Get the most recent snapshot, or a specific version.
      * @param specifiedVersion - The specific version of the snapshot to retrieve
-     * @returns The snapshot requested, or the latest snapshot if no version was specified
+     * @returns The snapshot requested, or the latest snapshot if no version was specified, plus version ID
      */
-    private async fetchSnapshotTree(specifiedVersion?: string): Promise<ISnapshotTree | undefined> {
+    private async fetchSnapshotTree(specifiedVersion: string | undefined | null):
+        Promise<[ISnapshotTree | undefined, string | undefined]>
+    {
+        if (specifiedVersion === null) {
+            return [undefined, undefined];
+        }
         const version = await this.getVersion(specifiedVersion ?? this.id);
 
-        if (version !== undefined) {
-            this._loadedFromVersion = version;
-            return await this.storageService.getSnapshotTree(version) ?? undefined;
-        } else if (specifiedVersion !== undefined) {
+        if (version === undefined && specifiedVersion !== undefined) {
             // We should have a defined version to load from if specified version requested
-            this.logger.sendErrorEvent({ eventName: "NoVersionFoundWhenSpecified", specifiedVersion });
+            this.logger.sendErrorEvent({ eventName: "NoVersionFoundWhenSpecified", id: specifiedVersion });
         }
+        this._loadedFromVersion = version;
+        const tree = await this.storageService.getSnapshotTree(version) ?? undefined;
```

Comment:
I believe there is a functional change here, is it correct?
If `version === undefined && specifiedVersion !== undefined` we will now do a call to `storageService.getSnapshotTree(undefined)`. Previously we would avoid this call and directly return undefined for the tree.

I feel like the logic here has become slightly more complicated. Other than adding logging, what is the purpose of the changes (except for the first `specifiedVersion === null` check)?

Reply:
Correct. But there is no longer a way to load from ops with detached container creation. So if version was specified but it does not exist, then we either fail (throw from here), or we find safe next best thing. returning tree=undefined is not it, loading from latest is safe(r).

Reply:
To answer your question - prior code was not exposing version ID and was logging tree ID (which are the same).
ID on tree is optional, and driver is free no to provide it, and this is very likely to happen with upcoming optimization work in ODSP driver. And possible I'll remove it from ISnapshotTree completely (not sure about that - it all depends on if we want to support in future partial trees in snapshots, and ability to request from storage sub-trees - there is no such API today on driver / in storage). 

So I want to prep this code not to depend on tree.id - just use version that we always get - it's part of contract, and expose it to load() to record in telemetry.

Reply:
Let me know what you think, happy to simplify if you see a way :)